contracts/libs/Witnet.sol

```diff
@@ -3,314 +3,371 @@
-  /// @return The `int128[]` decoded from the WitnetData.Result.
-  function asFixed16Array(WitnetData.Result memory _result) external pure returns(int32[] memory) {
-    require(_result.success, "Tried to read `fixed16[]` value from errored WitnetData.Result");
-    return _result.value.decodeFixed16Array();
-  }
-
-  /// @notice Decode a integer numeric value from a WitnetData.Result as an `int128` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `int128` decoded from the WitnetData.Result.
-  function asInt128(WitnetData.Result memory _result) external pure returns(int128) {
-    require(_result.success, "Tried to read `int128` value from errored WitnetData.Result");
-    return _result.value.decodeInt128();
-  }
-
-  /// @notice Decode an array of integer numeric values from a WitnetData.Result as an `int128[]` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `int128[]` decoded from the WitnetData.Result.
-  function asInt128Array(WitnetData.Result memory _result) external pure returns(int128[] memory) {
-    require(_result.success, "Tried to read `int128[]` value from errored WitnetData.Result");
-    return _result.value.decodeInt128Array();
-  }
-
-  /// @notice Decode a string value from a WitnetData.Result as a `string` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `string` decoded from the WitnetData.Result.
-  function asString(WitnetData.Result memory _result) external pure returns(string memory) {
-    require(_result.success, "Tried to read `string` value from errored WitnetData.Result");
-    return _result.value.decodeString();
-  }
-
-  /// @notice Decode an array of string values from a WitnetData.Result as a `string[]` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `string[]` decoded from the WitnetData.Result.
-  function asStringArray(WitnetData.Result memory _result) external pure returns(string[] memory) {
-    require(_result.success, "Tried to read `string[]` value from errored WitnetData.Result");
-    return _result.value.decodeStringArray();
-  }
-
-  /// @notice Decode a natural numeric value from a WitnetData.Result as a `uint64` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `uint64` decoded from the WitnetData.Result.
-  function asUint64(WitnetData.Result memory _result) external pure returns(uint64) {
-    require(_result.success, "Tried to read `uint64` value from errored WitnetData.Result");
-    return _result.value.decodeUint64();
-  }
+    /// Data kept in EVM-storage for every Request posted to the Witnet Request Board.
+    struct Request {
+        IWitnetRequest addr;    // The contract containing the Data Request which execution has been requested.
+        address requestor;      // Address from which the request was posted.
+        bytes32 codehash;       // Codehash of the Data Request which execution has been requested.
```

Comment:
For simplicity:
```suggestion
        bytes32 hash;       // Hash of the Data Request whose execution has been requested.
```

On `whose` vs. `which`: https://thewritepractice.com/whose-vs-which-inanimate-objects/

Reply:
Thanks for the grammar tip :-) 👍